pkg/logql/metrics.go

```diff
@@ -581,6 +626,95 @@ func extractShard(shards []string) *astmapper.ShardAnnotation {
 	return &shard
 }
 
+// calculateResultSize calculates an approximate estimate of the result size in bytes
+// without serialization by summing up the actual data sizes plus estimated JSON overhead.
+// This is an approximation and may not match the exact serialized size. Used for frontend logging.
+func calculateResultSize(result promql_parser.Value) int {
+	if result == nil {
+		return 0
+	}
+
+	switch v := result.(type) {
+	case logqlmodel.Streams:
+		var size int
+		for _, stream := range v {
+			// Stream labels
+			size += len(stream.Labels)
+			// JSON overhead for stream object (~20 bytes: {"stream":{},"values":[]})
+			size += 20
+			for _, entry := range stream.Entries {
+				// Entry line content
+				size += len(entry.Line)
+				// Timestamp as string (~20 bytes for RFC3339Nano)
+				size += 20
+				// JSON overhead for entry array (~10 bytes: ["timestamp","line"])
+				size += 10
+				// Structured metadata
+				for _, label := range entry.StructuredMetadata {
+					size += len(label.Name) + len(label.Value) + 10 // +10 for JSON overhead
+				}
+				// Parsed labels
+				for _, label := range entry.Parsed {
+					size += len(label.Name) + len(label.Value) + 10 // +10 for JSON overhead
+				}
+			}
+		}
```

Comment:
A comment every other line is really interrupting the reading flow here. Can we inline more of these and remove any of them which are obvious?

```go
size += len(stream.Labels)
size += 20 // JSON overhead for stream object: {"stream":{},"values":[]}
```

(etc)